### PR TITLE
Docs: updates component level docs

### PR DESCRIPTION
**Pull request by ZPain8464**
Updates component level docs. I'd like to get @calebdoxsey and @wasaga to take a look and offer any suggestions or corrections as they see fit. Databroker, for instance, could be fleshed out more. 

The diagram I'm working on will incorporate some of these steps as well.

### CODE REVIEW

File: content/docs/internals/architecture.md [hunk 1]
```diff
@@ -11,38 +11,46 @@ Pomerium sits between end users and services requiring strong authentication. Af
...
-  - All user traffic flows through the proxy
-  - Verifies all requests with Authentication service
-  - Directs users to Authentication service to establish session identity
-  - Processes policy to determine external/internal route mappings
-  - Receives user OIDC tokens and session data from Authentication service and stores in Databroker service
+The following steps outline how the Proxy service handles a request at a high level:
 
-- Authentication Service
+1. The client makes a request to access the target resource 
+1. The Proxy service receives the request and redirects it to the Authorization service before proxying it
```
Comment (calebdoxsey) [inline content/docs/internals/architecture.md:24]: it does not redirect it to the Authorization service. It makes a gRPC call to the Authorization service which evaluates policy.

File: content/docs/internals/architecture.md [hunk 2]
```diff
@@ -11,38 +11,46 @@ Pomerium sits between end users and services requiring strong authentication. Af
...
+1. If the Authorization service doesn’t see a session cookie, the Proxy service redirects the request to the Authentication service to verify the client’s identity
+1. After the Authentication service verifies the client’s identity and saves a local session cookie, the Proxy service redirects the session data to the Databroker service
```
Comment (calebdoxsey) [inline content/docs/internals/architecture.md:26]: it does not redirect the session data to the Databroker service. It makes a gRPC call to the Databroker service to save the session.

File: content/docs/internals/architecture.md [hunk 3]
```diff
@@ -11,38 +11,46 @@ Pomerium sits between end users and services requiring strong authentication. Af
...
-  - Handles authentication flow to your IdP as needed
-  - Handles identity verification after initial Authentication
-  - Establishes user session cookie
-  - Stores user OIDC tokens in Proxy service
+### Authentication service
+The Authentication service is responsible for authenticating users against an Identity Provider (IdP) and establishing user sessions. By incorporating OAuth 2.0 and OIDC protocols into the authentication flow, the Authentication service provides single sign-on authentication that enables it to extract user identity details and session data necessary for managing Pomerium sessions. 
 
-- Authorization Service
+At a high level, when the Authentication service first receives a request from the Proxy service, it:
+1. Sends an HTTP request on behalf of the client to the IdP
```
Comment (calebdoxsey) [inline content/docs/internals/architecture.md:33]: It redirects the user to the identity provider.

File: content/docs/internals/architecture.md [hunk 4]
```diff
@@ -11,38 +11,46 @@ Pomerium sits between end users and services requiring strong authentication. Af
...
+1. Completes the authentication flow with the IdP and extracts relevant session data, such as device credentials, user ID, issuance and expiration times, OAuth tokens, and any OIDC claims (scopes) provided by the IdP
+1. Saves session data to a local session cookie and redirects the client with session data encrypted in URL parameters to the Proxy service
+1. Verifies user identity after the session expires
```
Comment (calebdoxsey) [inline content/docs/internals/architecture.md:36]: After a session expires its as if the user is no longer logged in. Authenticate merely runs the login process again. It doesn't verify the user identity.

File: content/docs/internals/architecture.md [hunk 5]
```diff
@@ -11,38 +11,46 @@ Pomerium sits between end users and services requiring strong authentication. Af
...
-  - Processes policy to determine permissions for each service
-  - Handles authorization check for all user sessions
-  - Directs Proxy service to initiate Authentication flow as required
-  - Provides additional security related headers for upstream services to consume
+### Authorization service
+The Authorization service processes policies to determine what permissions the client has. Each request sent by the client must first go through an authorization check before the Proxy service proxies the request. 
 
-- Databroker Service
+When the Authorization service receives a request from the Proxy service, the following actions take place:
+1. The Authorization service first looks for a session cookie, which contains the client’s JWT. If no session cookie is present, the Authorization service sends a request to the Authentication service to establish session identity.
```
Comment (calebdoxsey) [inline content/docs/internals/architecture.md:42]: The Authorization service returns a Redirect response. The browser makes the request to the Authenticate service.

File: content/docs/internals/architecture.md [hunk 6]
```diff
@@ -11,38 +11,46 @@ Pomerium sits between end users and services requiring strong authentication. Af
...
-  - Handles authorization check for all user sessions
-  - Directs Proxy service to initiate Authentication flow as required
-  - Provides additional security related headers for upstream services to consume
+### Authorization service
+The Authorization service processes policies to determine what permissions the client has. Each request sent by the client must first go through an authorization check before the Proxy service proxies the request. 
 
-- Databroker Service
+When the Authorization service receives a request from the Proxy service, the following actions take place:
+1. The Authorization service first looks for a session cookie, which contains the client’s JWT. If no session cookie is present, the Authorization service sends a request to the Authentication service to establish session identity. 
+1. Once a session cookie is in place, the Proxy service redirects the request to the Authorization service so it can determine permissions based on JWT claims and policy.
```
Comment (calebdoxsey) [inline content/docs/internals/architecture.md:43]: The Proxy service does not redirect the request to the Authorization service. It makes a gRPC call.

File: content/docs/internals/architecture.md [hunk 7]
```diff
@@ -11,38 +11,46 @@ Pomerium sits between end users and services requiring strong authentication. Af
...
-  - Provides additional security related headers for upstream services to consume
+### Authorization service
+The Authorization service processes policies to determine what permissions the client has. Each request sent by the client must first go through an authorization check before the Proxy service proxies the request. 
 
-- Databroker Service
+When the Authorization service receives a request from the Proxy service, the following actions take place:
+1. The Authorization service first looks for a session cookie, which contains the client’s JWT. If no session cookie is present, the Authorization service sends a request to the Authentication service to establish session identity. 
+1. Once a session cookie is in place, the Proxy service redirects the request to the Authorization service so it can determine permissions based on JWT claims and policy. 
+1. The Authorization service then constructs security headers based on JWT claims, which the Proxy service forwards to upstream applications.
+1. With each subsequent request, the Authorization service checks the session expiration time and communicates session changes to the Databroker.
```
Comment (calebdoxsey) [inline content/docs/internals/architecture.md:45]: The Authorization service does not communicate session changes to the Databroker.

File: content/docs/internals/architecture.md [hunk 8]
```diff
@@ -11,38 +11,46 @@ Pomerium sits between end users and services requiring strong authentication. Af
...
-  - Retrieves identity provider related data such as group membership
-  - Stores and refreshes identity provider access and refresh tokens
-  - Provides streaming authoritative session and identity data to Authorize service
-  - Stores session and identity data in persistent storage
+### Databroker service
+The Databroker service persists session and identity-related data and streams session state updates to other services. It also functions as an identity manager in that it's responsible for refreshing user sessions against the IdP until a Pomerium session has expired. 
+
+The points below outline the Databroker’s role in the request and session lifecycle:
+
+- Once a client is authenticated, the Proxy server redirects the session data and identity information to the Databroker. The Databroker persists the data.
```
Comment (calebdoxsey) [inline content/docs/internals/architecture.md:52]: The Proxy service does not redirect the session data and identity information. It makes a gRPC call to the Databroker to save session data.

File: content/docs/internals/architecture.md [hunk 9]
```diff
@@ -11,38 +11,46 @@ Pomerium sits between end users and services requiring strong authentication. Af
...
-  - Stores and refreshes identity provider access and refresh tokens
-  - Provides streaming authoritative session and identity data to Authorize service
-  - Stores session and identity data in persistent storage
+### Databroker service
+The Databroker service persists session and identity-related data and streams session state updates to other services. It also functions as an identity manager in that it's responsible for refreshing user sessions against the IdP until a Pomerium session has expired. 
+
+The points below outline the Databroker’s role in the request and session lifecycle:
+
+- Once a client is authenticated, the Proxy server redirects the session data and identity information to the Databroker. The Databroker persists the data. 
+- The Databroker streams authoritative session and identity data to the Authorization service, keeping the two services in sync.
```
Comment (calebdoxsey) [inline content/docs/internals/architecture.md:53]: The Databroker no longer streams authoritative session and identity data to the Authorization service. That data is queried on-demand by the Authorization service.

**Review by calebdoxsey — CHANGES_REQUESTED** [review]

**Review by calebdoxsey — APPROVED** [review]

**Review by cmo-pomerium — COMMENTED** [review]
LGTM

**Pull request merged**